### ISSUE TITLE
Experience levels

### ISSUE THREAD

**Issue by MichaelMackus** [LambdaHack/LambdaHack#234]
It would be good to have a way to implement experience levels in the game (for more traditional RPG roguelikes).

See [wiki page](https://github.com/LambdaHack/LambdaHack/wiki/Experience-levels) for more detailed design idea

**Comment by Mikolaj** [comment]
Yes, let's brainstorm. Personally, I'd like to check how much can be done already exclusively in content. If all, we are set. If not, I'd prefer to extend the engine minimally to let this be done in content. Then, after some content writing tests and playtests, after it stabilizes, we could extend the engine some more to improve comfort of writing content and improve the presentation to the player.

BTW, you may want to skim through https://www.reddit.com/r/roguelikedev/comments/9ewew0/advice_needed_how_to_gain_xp/. I also have 370 lines of notes about that (completely cryptic, but I share via email). I've implemented a lot of groundwork for that in terms of various levels of existing stats (the first few of the `Skill`) giving various boons (so that XP increasing stats is interesting; a lot more work is needed and also stats can be added) and recording kills and a lot of other actions for each actor of each faction (`ActorAnalytics`) and for each faction in total (`FactionAnalytics`) and about things found and experiences ('GenerationAnalytics') so that XP can be granted whenever the records pass various tresholds (I guess there is enough analytics stored already in each game to have a good start). You can see a modest portion of the analytics among the game over screens, particularly when running with options `--exposeItems --exposeActors --showItemSamples`.

I suspect not many engine changes will be needed, but a lot of gameplay design (e.g., flesh out the stats and think of other ways XP gets interesting) and content writing.
